core/appeal/service.go

```diff
@@ -525,6 +529,135 @@ func (s *Service) Revoke(ctx context.Context, id string, actor, reason string) (
+	}
+
+	if errs := s.notifier.Notify([]domain.Notification{
+		{
+			User: email,
+			Message: domain.NotificationMessage{
+				Type: domain.NotificationTypeApproverNotification,
+				Variables: map[string]interface{}{
+					"resource_name": fmt.Sprintf("%s (%s: %s)", appeal.Resource.Name, appeal.Resource.ProviderType, appeal.Resource.URN),
+					"role":          appeal.Role,
+					"requestor":     appeal.CreatedBy,
+					"appeal_id":     appeal.ID,
+				},
+			},
+		},
+	}); errs != nil {
+		for _, err1 := range errs {
+			s.logger.Error("failed to send notifications", "error", err1.Error())
+		}
+	}
+
+	return appeal, nil
+}
+
+func (s *Service) DeleteApprover(ctx context.Context, appealID, approvalID, email string) (*domain.Appeal, error) {
+	if err := s.validator.Var(email, "email"); err != nil {
+		return nil, fmt.Errorf("%w: %s", ErrApproverEmail, err)
+	}
+
+	appeal, approval, err := s.getApproval(appealID, approvalID)
+	if err != nil {
+		return nil, err
+	}
+	if appeal.Status != domain.AppealStatusPending {
+		return nil, fmt.Errorf("%w: can't delete approver to appeal with %q status", ErrUnableToDeleteApprover, appeal.Status)
+	}
+
+	if !utils.ContainsString([]string{domain.ApprovalStatusPending, domain.ApprovalStatusBlocked}, approval.Status) {
+		return nil, fmt.Errorf("%w: can't delete approver to approval with %q status", ErrUnableToDeleteApprover, approval.Status)
+	}
+
+	// check if approval type is auto
+	// this approach is the quickest way to assume that approval is auto, otherwise need to fetch the policy details and lookup the approval type which takes more time
+	if approval.Status == domain.ApprovalStatusBlocked && (approval.Approvers == nil || len(approval.Approvers) == 0) {
+		// approval is automatic (strategy: auto) that is still on blocked
+		return nil, fmt.Errorf("%w: can't modify approvers for approval with strategy auto", ErrUnableToDeleteApprover)
+	}
+
+	if len(approval.Approvers) == 1 {
+		return nil, fmt.Errorf("%w: can't delete if there's only one approver", ErrUnableToDeleteApprover)
+	}
+
+	if err := s.approvalService.DeleteApprover(ctx, approvalID, email); err != nil {
+		return nil, err
+	}
+
+	var newApprovers []string
+	for _, a := range approval.Approvers {
+		if a != email {
+			newApprovers = append(newApprovers, a)
+		}
+	}
+	approval.Approvers = newApprovers
+
+	if err := s.auditLogger.Log(ctx, AuditKeyDeleteApprover, approval); err != nil {
+		s.logger.Error("failed to record audit log", "error", err)
+	}
+
+	return appeal, nil
+}
+
+func (s *Service) getApproval(appealID, approvalID string) (*domain.Appeal, *domain.Approval, error) {
+	if appealID == "" {
+		return nil, nil, ErrAppealIDEmptyParam
+	}
+	if approvalID == "" {
+		return nil, nil, ErrApprovalIDEmptyParam
+	}
+
+	appeal, err := s.repo.GetByID(appealID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("getting appeal details: %w", err)
+	}
+
+	var approval *domain.Approval
+	for _, a := range appeal.Approvals {
```

Comment:
this could be move to appeal struct as a receiver function 
```go
func (ap *Appeal) GetApprovalByID(approvalID string) *domain.Approval {
	for _, a := range ap.Approvals {
		if a.ID == approvalID || a.Name == approvalID {
			return a
		}
	}
        return nil
}
```

later you could do
```go
if approval  := appeal.GetApprovalByID(approvalID); approval != nil {
  return appeal, approval, nil
}
return nil, nil, ErrApprovalNotFound
```